gcp_variant_transforms/vcf_to_bq_preprocess.py

```diff
@@ -0,0 +1,113 @@
+# Copyright 2018 Google Inc.  All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+r"""Pipeline for preprocessing the VCF files.
+
+This pipeline is aimed to help the user to easily identify and further import
+the malformed/incompatible VCF files to BigQuery. It generates two files as the
+output:
+- Conflicts report: A file that lists the incompatible headers, undefined header
+  fields, the suggested resolutions and eventually malformed records.
+- Resolved headers file: A VCF file that contains the resolved fields
+  definitions.
+
+The files are generated in the "temp_location" if it is provided. Otherwise,
+they will be saved in the ``directory``.
+
+Run locally:
+python -m gcp_variant_transforms.vcf_to_bq_preprocess \
+  --input_pattern <path to VCF file(s)>
+
+Run on Dataflow:
+python -m gcp_variant_transforms.vcf_to_bq_preprocess \
+  --input_pattern <path to VCF file(s)>
+  --report_all True \
+  --project gcp-variant-transforms-test \
+  --job_name preprocess \
+  --staging_location "gs://integration_test_runs/staging" \
+  --temp_location "gs://integration_test_runs/temp" \
+  --runner DataflowRunner \
```

Comment:
I guess default `directory`, and filenames are used if not specified. Right?

Reply:
I think asking the user for the full path on runs is a better option. That way, we don't add any random prefixes to their report names. We could add logic to make sure we don't overwrite existing files (lower priority though).
P.S. The reason we decided to write the random representative header in the main module was that the user did not ask for that file, but we had to generate it. This is not applicable here as the user is requesting that path (well, we ask them to :P)